mkosi/__init__.py

```diff
@@ -3339,6 +3339,14 @@ def make_image(
 
     logging.debug(json.dumps(output, indent=4))
 
+    if context.config.verity == Verity.defer and ArtifactOutput.roothash in context.config.split_artifacts:
```

Comment:
This should be independent of whether `defer` is used or not, if the user want the roothash as a split artifact, give it to him!